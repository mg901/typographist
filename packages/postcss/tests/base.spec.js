const run = require('./run');
const { userConfig } = require('../../../mocks');

describe('@base', () => {
  it(
    'should replace @base with css variables with breakpoint values, ' +
      'the font size specified in percent, specify the font size for each breakpoint',
    () => {
      const source = `
            body {
              @base;
            }`;
      const compiled = `
            body {
              font-size: 1.3333333333333333rem;
              line-height: 2rem;
              font-style: normal;
              font-weight: normal;
            }
@media (min-width: 48em) {
              body {
                            font-size: 1.3076923076923077rem;
              }
}
@media (min-width: 62em) {
              body {
                            font-size: 1.3103448275862069rem;
              }
}
@media (min-width: 75em) {
              body {
                            font-size: 1.3125rem;
              }
}`;
      return run(source, compiled, userConfig);
    },
  );
});
