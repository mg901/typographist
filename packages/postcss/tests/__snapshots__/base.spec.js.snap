// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base transform @base 1`] = `
body {
  font-size: 1.3333333333333333rem;
  line-height: 2rem;
  font-style: normal;
  font-weight: normal;
}
@media (min-width: 48em) {
  body {
    font-size: 1.3076923076923077rem;
  }
}
@media (min-width: 62em) {
  body {
    font-size: 1.3103448275862069rem;
  }
}
@media (min-width: 75em) {
  body {
    font-size: 1.3125rem;
  }
}
`;
